pandas/core/indexing.py

```diff
@@ -675,8 +676,22 @@ def _ensure_listlike_indexer(self, key, axis=None, value=None):
             # GH#38148
             keys = self.obj.columns.union(key, sort=False)
 
+            # Try to get the right dtype when we do this reindex.
```

Comment:
@phofl any ideas on how to do this more effectively?  in particular, to get the 32 bit builds passing

Reply:
Concerning the 32bit builds: I think we should only infer the fv if value has a dtype or is a scalar (think is the important part here, not quite sure what to expect with lists)